spark/src/main/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuery.scala

```diff
@@ -352,23 +352,23 @@ trait OptimizeMetadataOnlyDeltaQuery extends LoggingShims {
       case _ => false
     }
 
-    def unapply(plan: Aggregate): Option[TahoeLogFileIndex] = plan match {
-      case Aggregate(
-        Nil, // GROUP BY not supported
-        aggExprs: Seq[Alias @unchecked], // Underlying type is not checked because of type erasure.
-        // Alias type check is done in isStatsOptimizable.
-        PhysicalOperation(fields, Nil, DeltaTable(fileIndex: TahoeLogFileIndex)))
-          if fileIndex.partitionFilters.isEmpty &&
-            fieldsAreAttributeReference(fields) &&
-            isStatsOptimizable(aggExprs) => Some(fileIndex)
-      case Aggregate(
-        Nil,
-        aggExprs: Seq[Alias @unchecked],
-        // When all columns are selected, there are no Project/PhysicalOperation
-        DeltaTable(fileIndex: TahoeLogFileIndex))
-          if fileIndex.partitionFilters.isEmpty &&
-            isStatsOptimizable(aggExprs) => Some(fileIndex)
-      case _ => None
+    def unapply(plan: Aggregate): Option[TahoeLogFileIndex] = {
+      // GROUP BY is not supports. All AggregateExpression must be stats optimizable.
```

Comment:
Needed because of https://github.com/apache/spark/commit/d6b733492d3e1643babbcd2b1860e35af7928906